image-prune

```diff
@@ -199,6 +202,30 @@ class LocalImageDirectory(ImageCache):
         return free >= self.space_threshold
 
 
+class S3ImageStore(ImageCache):
+    def __init__(self, url, max_gb=80):
```

Comment:
TBH this feels a bit too arbitrary and too well hidden in the code. Can you please at least put it as a `s3.IMAGES_MAX_GB` constant at the top of the file or into lib/s3.py with a comment? This makes assumptions about how many images we have at any given time. It should be fine, but I'd like to make it a bit more explicit.

I doesn't really impedance-match what the local stores are doing with the "remaining space pressure", but as S3 works differently I dont' have a better idea either.

Reply:
Sure.

The issue here is that S3 has no reasonable upper bounds on storage space: we use more we pay more.  Therefore "free space" is kinda meaningless.  It gets even more confusing when we consider that the 250GB included in the base price is split between all of our configured buckets, but we only have a view on a single bucket at a time.

The chosen value of 80 corresponds to the fact that we have two buckets, and keeps us very comfortably below our 250GB "included in base price" quota.  We could probably even make it 100, but that's bikeshedding.

Reply:
I changed it around to match what we do with the local prune threshold (envvar) and added a comment explaining the logic.  I'd prefer not to add a constant to the top of the file because it's S3-specific, and I'd prefer not to add a constant to `lib.s3` because it's prune-specific.

Reply:
Right, that was just thinking aloud. That's what I meant with "S3 works differently". The approach is fine, it's just a bit hard to see when you look through the code. Personally I prefer having such magic tweakable constants at the top of the files, and function default parameters shoudl just be something like `=None` or `={}`.